### PR TITLE
Neo 53889 adding abortable support

### PR DESCRIPTION
**Pull request by infinity1013**
## Description

Currently there is no way to abort a request from the client. For example:

UI makes requests to fetch the data
While the call is in progress, the UI component unmounts/redirects to some other component.
While un-mounting, the UI should be able to cancel pending HTTP calls to avoid unnecessary attempts to re-render.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,3229 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (gauravmakkar) [inline yarn.lock:1]: please remove this file. js sdk uses package-lock.json

File: src/soap.js [hunk 2]
```diff
@@ -575,7 +576,7 @@ class SoapMethodCall {
      * @param {string} url the endpoint (/nl/jsp/soaprouter.jsp)
      * @param {client.Client} sdk client (optional)
      */
```
Comment (gauravmakkar) [inline src/soap.js:578]: jsdoc needs to be updated for signal

File: src/client.js [hunk 3]
```diff
@@ -1185,7 +1185,7 @@ class Client {
      * @param {Campaign.XtkMethodParam[]}  outputParams is the list of output parameters. The first paramater in the array is the "this" parameter for non-static method calls
      * @param {string} representation is the representation to use to interpret the parameters 
      */
```
Comment (gauravmakkar) [inline src/client.js:1187]: jsdoc needs to be updated for signal

**Review by gauravmakkar — CHANGES_REQUESTED** [review]

**Comment by gauravmakkar** [comment]
@infinity1013 don't you think the docs should be updated for the usage of this feature? It should be added even in the best practises as well.

File: src/transport.js [hunk 4]
```diff
@@ -114,6 +114,7 @@ if (!Util.isBrowser()) {
         method: options.method,
         headers: headers,
         body: options.data,
+        ...(options.signal && {signal : options.signal})
```
Comment (mkiki) [inline src/transport.js:117]: Why not write "signal: options.signal" ?
The SDK does not currently use any build tools, and hence trying to keep the syntax as simple as possible.

File: src/soap.js [hunk 5]
```diff
@@ -554,7 +554,8 @@ class SoapMethodCall {
             url: url,
             method: 'POST',
             headers: headers,
-            data: DomUtil.toXMLString(this._doc)
+            data: DomUtil.toXMLString(this._doc),
+            ...(this._pushDownOptions && {signal : this._pushDownOptions.signal})
```
Comment (mkiki) [inline src/soap.js:558]: I do not think this is necessary. The pushdown mechanism should pass the signal attirbute automatically down to the transport layer

File: src/transport.js (hunk 4 above)
Comment (mkiki) [reply src/transport.js:117]: Also, what about the axis transport layer? Both fetch and axis transport layers should be updated

File: docs/transport.html [hunk 6]
```diff
@@ -23,7 +23,7 @@
 
 </ul>
 
-<p>The second parameter is an set of additional parameters that have been pushed down to the transport layer (see the <b>Pushdown</b> paragraph for more details). In particular, the <b>timeout</b> parameter should be honored by the transport layer.</p>
+<p>The second parameter is an set of additional parameters that have been pushed down to the transport layer (see the <b>Pushdown</b> paragraph for more details). In particular, the <b>timeout</b> and <b>AbortSignal</b> parameter should be honored by the transport layer.</p>
```
[the remark this answers is not in the document]
Comment (mkiki) [reply docs/transport.html:26]: Can you also document how to abort requests? Look at the timeout.html for an example.
Add a new "Aborting queries" menu in Advanced topics section, just like this is done for timeouts. Document how the mechansim works and give a few exampels.

**Review by mkiki — CHANGES_REQUESTED** [review]

File: docs/abortRequest.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (mkiki) [inline docs/abortRequest.html:1]: You also need to update navigation.yml to make sure this new page appears in the doc menu. You can add an entry right after the timeouts.html entry

File: docs/abortRequest.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+layout: page
+title: Handling abort requests
+---
+
+<p>When building applications, we make API requests to servers for different reasons; to authenticate user credentials, fetch resources, create resources, and so much more. In JavaScript,we can abort an API request i.e cancel a request before it is completed. There are many reasons why you may want to do this. One of them is to prevent a request from being sent to the server if the user has already navigated away from the page the request was made. Canceling requests can help you avoid unnecessary network traffic, save memory usage and resources, and improve performance.</p>
+
+<h3>Aborting API requests in fetch</h3>
```
Comment (mkiki) [inline docs/abortRequest.html:8]: You should start with &lt;h1> for top level titles, not &lt;h3>

File: docs/abortRequest.html [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+<p>Here's the flow of how canceling a fetch call works:</p>
+<ul>
+    <li>Create an <b>AbortController</b> instance</li>
+    <li>That instance has a <b>signal</b> property</li>
+    <li>Pass the signal as a fetch option for signal</li>
+    <li>Call the AbortController's abort property to cancel all fetches that use that signal.</li>
+</ul>
+
+<h3>Aborting API requests in Axios</h3>
```
Comment (mkiki) [inline docs/abortRequest.html:20]: You should start with &lt;h1> for top level titles, not &lt;h3>

File: docs/abortRequest.html [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+<p>Axios is a JavaScript library that is used to make API requests. It has a very similar syntax to the fetch method. It also accepts a signal property in the request options object. When the aborted flag of the signal object is updated to true, axios expression is notified, and the request is canceled.</p>
+
+<h2>Passing AbortSignal via PushDown Mechanism</h2>
```
Comment (mkiki) [inline docs/abortRequest.html:24]: You should start with &lt;h1> for top level titles, not &lt;h3>

File: docs/abortRequest.html [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+<p>SDK provides the support to pass the signal directly to transport layer via <b>PushDown Mechanism</b></p>
+
+<pre class="code">
+client.NLWS.pushDown({ signal: instance.current.signal }).xtkQueryDef.create(queryDef)
```
Comment (mkiki) [inline docs/abortRequest.html:29]: I think the example could show the creation of the signal object and how to actually abort the request

**Review by mkiki — CHANGES_REQUESTED** [review]
Hi Aadarsh, there are a few details in the documentation, and then I think it's all good.

**Pull request merged**